README.md

```diff
@@ -396,6 +397,74 @@ end
 
 The Dashboard can be set to automatically refresh by checking "Live Poll" in the Dashboard header, or by setting `?poll=10` with the interval in seconds (default 30 seconds).
 
+### Batches
+
+Batches track a set of jobs, and enqueue an optional callback job when all of the jobs have finished (succeeded or discarded).
+
+- A simple example that enqueues your `MyBatchCallbackJob` after the two jobs have finished, and passes along the current user as a batch property:
+
+    ```ruby
+    GoodJob::Batch.enqueue(MyBatchCallbackJob, user: current_user) do
+      MyJob.perform_later
+      OtherJob.perform_later
+    end
+
+    # When these jobs have finished, it will enqueue your `MyBatchCallbackJob.perform_later(batch, options)`
+    class MyBatchCallbackJob < ApplicationJob
+      def perform(batch, options = {})
+        # Do something with the <GoodJob::Batch> instance; options is a hash is reserved for future use
+        batch.properties[:user] # => <User id: 1, ...>
+      end
+    end
+    ```
+
+- Jobs can be added to an existing batch. Jobs in a batch are enqueued and performed immediately, though the final callback job will not be enqueued until `GoodJob::Batch#enqueue` is called.
+
+    ```ruby
+    batch = GoodJob::Batch.add do
+      10.times { MyJob.perform_later }
+    end
+    batch.add do
+      10.times { OtherJob.perform_later }
+    end
+    batch.enqueue(MyBatchCallbackJob, age: 42)
+    ```
+
+- If you need to access the batch within a model, include [`GoodJob::ActiveJobExtensions::Batches`](lib/good_job/active_job_extensions/batches.rb) in your job class:
```

Comment:
Am I allowed to add jobs dynamically to the batch? (If yes, let's add an example in the documentation, this is extremely useful.)